tests/test_monitor.py

```diff
@@ -364,24 +364,19 @@ def test_compound_fail(self):
             "compound monitor did not report failures properly",
         )
 
-    def test_recovery(self):
+    @mock.patch("subprocess.Popen")
+    def test_recovery(self, mock_popen):
         m = MonitorFail("fail1", {"recover_command": "touch did_recovery"})
-        try:
-            os.unlink("did_recovery")
-        except FileNotFoundError:
-            pass
         m.run_test()
         m.attempt_recover()
-        #  throws an exception if the file isn't there
-        os.stat("did_recovery")
+        # Make sure we called the fake command
+        mock_popen.assert_called_once_with(["touch", "did_recovery"])
```

Comment:
note: I kept the command and the arguments as they were. But since we're not calling it, this really could be anything (and could use more complex args if we want to make sure argument parsing / etc. is handled as expected.